apps/meteor/server/services/moderation/service.ts

```diff
@@ -0,0 +1,48 @@
+import { ServiceClassInternal } from '@rocket.chat/core-services';
+import type { IModerationService } from '@rocket.chat/core-services';
+import { Permissions, Roles, Users } from '@rocket.chat/models';
+
+import { createOrUpdateProtectedRoleAsync } from '../../lib/roles/createOrUpdateProtectedRole';
+
+export class ModerationService extends ServiceClassInternal implements IModerationService {
+	protected name = 'moderation';
+
+	async resetUserRoles(roles: string[]): Promise<void> {
+		const usersInRole = await Users.findUsersInRoles(roles, undefined, { projection: { _id: 1 } }).toArray();
+
+		const usersToRemove = usersInRole.map((user) => user._id);
+
+		// remove the roles from the users
+		await Roles.removeUsersRoles(usersToRemove, roles);
+
+		// add the default 'user' role to the users
+		await Roles.addRolesByUserIds(usersToRemove, ['user']);
```

Comment:
not all workspace defaults to user role. this should be skipped. as when the trust roles are added, we're not unsetting the default role either. upto the admin to change the default role.

Reply:
The app will unset the default roles when installed; if it is not done, then the system won't work because usually, the permissions check is done considering the highest level role.
For example, if a user has two roles, `novice` and `users`, then there won't be any restriction, right? Because when checking for permission, the `users` will be given the preference. 
Please correct me if I misunderstood something. Thank you!

Reply:
Ok. But user is not always the default role. You need to reset to the default role and not to "user" specifically.

Reply:
When the setting is disabled, it'll then be safe to reset roles to present default ones.

Reply:
Ok, I removed the part which adds the `users` role.